Moving import of visualization library inside of function

senickel opened the pull request:
This will prevent the installation of additional packages on linux when not using visualizations (e.g., when working running code in a container).

jonnymaserati reviewed (changes requested):
@senickel thanks for the request and good spot... I've gotten a bit lazy recently and appreciate your feedback!

Suggest you review the PEP 8 guidelines on imports [here](https://peps.python.org/pep-0008/#imports). I think a nicer way to handle this would be to import the `PLOTLY` variable from `visual` and then:

```python
if PLOTLY:
    return figure(self, type, **kwargs)
else:
    return None
```
Or perhaps `print("Please pip install plotly to enable figures")` or something similar to prompt this dependency be installed?

jonnymaserati reviewed (changes requested):
Think you need to update your `.gitignore` file - these files should be in the repo.

jonnymaserati reviewed (changes requested):
Same comment as previous - suggest `from .visual import PLOTLY` and to conditionally return the `fig`.

jonnymaserati reviewed (changes requested):
Same comment here, make it conditional on `visual.PLOTLY`.

jonnymaserati reviewed (commented):
Yes! Good spot @nicolaphee. I've been flip flopping between `vedo` and `vtk` and was too lazy to add this. This needs cascading through the module though.

jonnymaserati reviewed (changes requested):
How to nicely handle when `VEDO` is `False`?

jonnymaserati reviewed (changes requested):
How to manage `if not VTK`?

jonnymaserati reviewed (changes requested):
Good stuff, but it would be nice to inform the user that for this functionality `pip install vtk`, either when the library is initiated or when the `class` is called?

jonnymaserati commented:
Don't forget to bump the `__version__` number in `.version`.

senickel commented:
Hi @jonnymaserati!
I finally have some time to review your comments. Thanks a lot for responding and for the great library as well!

Somehow your comments are not referencing any particular code lines on my side. Hence, where I wasn't sure what the comment was about I added :eyes: 

I also changed `visual.py` to raise an ImportError if the vtk or vedo are not installed but are needed to execute something. 

Let me know what you think!

rafaelmaza commented:
thanks for the great library @jonnymaserati (and for the PR, @senickel!) 

we have the same need, and were wondering this is planned to be reviewed sometime in the near future? 🙏
